_posts/2024-08-19-neural-sparse-v2-models.md

```diff
@@ -0,0 +1,179 @@
+| [opensearch-neural-sparse-encoding-doc-v1](https://huggingface.co/opensearch-project/opensearch-neural-sparse-encoding-doc-v1) | 0.490 | 0.707 | 0.352 | 0.521 | 0.677 | 0.344 | 0.461 | 0.294 | 0.412 | 0.154 | 0.743 | 0.202 | 0.716 | 0.788 |
+| [opensearch-neural-sparse-encoding-doc-v2-distill](https://huggingface.co/opensearch-project/opensearch-neural-sparse-encoding-doc-v2-distill) | 0.504 | 0.690 | 0.343 | 0.528 | 0.675 | 0.357 | 0.496 | 0.287 | 0.418 | 0.166 | 0.818 | 0.224 | 0.715 | 0.841 |
+| [opensearch-neural-sparse-encoding-doc-v2-mini](https://huggingface.co/opensearch-project/opensearch-neural-sparse-encoding-doc-v2-mini) | 0.497 | 0.709 | 0.336 | 0.510 | 0.666 | 0.338 | 0.480 | 0.285 | 0.407 | 0.164 | 0.812 | 0.216 | 0.699 | 0.837 |
+
+## Registering and deploying v2 models
+
+OpenSearch now provides [pretrained v2 sparse encoding models](https://opensearch.org/docs/latest/ml-commons-plugin/pretrained-models/#sparse-encoding-models). Depending on the search mode, you need to register different models:
+
+- In **doc-only mode**, you need to register a sparse encoding model for ingestion and a tokenizer for search. 
+- In **bi-encoder mode**, you need to register a sparse encoding model that will be used for ingestion and search.
+
+For detailed setup steps and tutorials, see [Neural sparse search](https://opensearch.org/docs/latest/search-plugins/neural-sparse-search/).
+
+### Registering and deploying models in doc-only mode
+
+To register and deploy models in doc-only mode, use the following steps.
+
+1. Register and deploy a sparse encoding model for ingestion:
+
+    ```json
+    POST /_plugins/_ml/models/_register?deploy=true
+    {
+        "name": "amazon/neural-sparse/opensearch-neural-sparse-encoding-doc-v2-distill",
+        "version": "1.0.0",
+        "model_format": "TORCH_SCRIPT"
+    }
+    ```
+
+1. Register and deploy a tokenizer for search:
+
+    ```
+    POST /_plugins/_ml/models/_register?deploy=true
+    {
+        "name": "amazon/neural-sparse/opensearch-neural-sparse-tokenizer-v1",
+        "version": "1.0.1",
+        "model_format": "TORCH_SCRIPT"
+    }
+    ```
+
+1. Get model IDs for the model and tokenizer by calling the Tasks API:
+
+    ```
+    GET /_plugins/_ml/tasks/{task_id}
+    ```
+
+### Registering and deploying models in bi-encoder mode
+
+To register and deploy models in bi-encoder mode, use the following steps.
+
+1. Register and deploy a sparse encoding model for ingestion and search:
+
+    ```json
+    POST /_plugins/_ml/models/_register?deploy=true
+    {
+        "name": "amazon/neural-sparse/opensearch-neural-sparse-encoding-v2-distill",
+        "version": "1.0.0",
+        "model_format": "TORCH_SCRIPT"
+    }
+    ```
+
+1. Get the model ID for the sparse encoding model by calling the Tasks API:
+
+    ```
```

Comment:
json?

Reply:
On the project side, when we don't have the actual request body, just the path and method will not be highlighted.